rpki-validator/src/main/java/net/ripe/rpki/validator3/api/roas/ExportsController.java

```diff
@@ -123,74 +121,114 @@ public void exportCsv(HttpServletResponse response) throws IOException {
+    public void exportCsvOnlyRoas(HttpServletResponse response) throws IOException {
+        response.setContentType(CSV);
+
+        if (!storage.readTx(settings::isInitialValidationRunCompleted)) {
+            response.setStatus(HttpServletResponse.SC_NO_CONTENT);
+            return;
+        }
 
-        Stream<JsonRoaPrefixExtended> validatedPrefixes = validatedRpkiObjects
-                .findCurrentlyValidatedRoaPrefixes()
-                .getObjects()
-                .map(r -> new JsonRoaPrefixExtended(
-                        String.valueOf(r.getAsn()),
-                        r.getPrefix().toString(),
-                        r.getEffectiveLength(),
-                        r.getTrustAnchor().getName(),
-                        Instant.ofEpochMilli(r.getNotBefore()).toString(),
-                        Instant.ofEpochMilli(r.getNotAfter()).toString(),
-                        r.getSerialNumber().toString()
-                ))
-                .distinct();
-
-        return new JsonExportExtended(validatedPrefixes);
+        try (CSVWriter writer = new CSVWriter(response.getWriter())) {
+            writer.writeNext(new String[]{"ASN", "IP Prefix", "Max Length", "Trust Anchor",
+                                          "Not Valid Before", "Not Valid After", "Serial Number"});
+            vrpExtendedStream().forEach(vrp ->
+                writer.writeNext(new String[]{
+                    vrp.getAsn(),
+                    vrp.getPrefix(),
+                    String.valueOf(vrp.getMaxLength()),
+                    vrp.getTa(),
+                    vrp.getNotBefore(),
+                    vrp.getNotAfter(),
+                    vrp.getSerialNumber()
+                }));
+        }
     }
 
-    @Value
-    private static class CsvRoaPrefix {
-        private String asn;
-        private String prefix;
-        private int maxLength;
-        private String trustAnchorName;
+    /**
+     * Get the stream of VRPs based only on ROAs, but having more fields.
+     */
+    private Stream<ExtendedRoa> vrpExtendedStream() {
+        return validatedRpkiObjects
+            .findCurrentlyValidatedRoaPrefixes()
+            .getObjects()
+            .map(r -> new ExtendedRoa(
+                String.valueOf(r.getAsn()),
+                r.getPrefix().toString(),
+                r.getEffectiveLength(),
+                r.getTrustAnchor().getName(),
+                Instant.ofEpochMilli(r.getNotBefore()).toString(),
+                Instant.ofEpochMilli(r.getNotAfter()).toString(),
+                r.getSerialNumber().toString()
+            ))
+            .distinct();
     }
 
+    /**
+     * Get the stream of VRPs based on ROAs - whitelist + SLURM assertions.
+     */
+    private Stream<VRP> vrpStream() {
+        Stream<VRP> validatedPrefixes = validatedRpkiObjects
+            .findCurrentlyValidatedRoaPrefixes()
+            .getObjects()
+            .filter(new IgnoreFiltersPredicate(ignoreFilters.all()).negate())
```

Comment:
Somehow we need to mark PR with refactoring where to focus on most relevant changes :P 

Here is what changed and makes difference on the whole story :)

Reply:
Yeah :)
Just had a fit of boyscouting.